packages/actor-rdf-dereference-file/lib/ActorRdfDereferenceFile.ts

```diff
@@ -0,0 +1,80 @@
+import {ActorRdfDereference, IActionRdfDereference, IActorRdfDereferenceOutput} from "@comunica/bus-rdf-dereference";
+import {
+  ActorRdfParse,
+  IActionRootRdfParse,
+  IActorOutputRootRdfParse,
+  IActorTestRootRdfParse,
+} from "@comunica/bus-rdf-parse";
+import {IActorArgs, IActorTest, Mediator} from "@comunica/core";
+import * as fs from "fs";
+import * as path from "path";
+import {URL} from "url";
+import {promisify} from "util";
+
+/**
+ * A comunica File RDF Dereference Actor.
+ */
+export class ActorRdfDereferenceFile extends ActorRdfDereference {
+
+  public readonly mediatorRdfParse: Mediator<ActorRdfParse,
+    IActionRootRdfParse, IActorTestRootRdfParse, IActorOutputRootRdfParse>;
+
+  public readonly mediaMappings: { [id: string]: string };
+
+  constructor(args: IActorRdfDereferenceFileArgs) {
+    super(args);
+  }
+
+  public async test(action: IActionRdfDereference): Promise<IActorTest> {
+    if (!action.url.startsWith("file:///")) {
+      // check if the url contains a (relative) path
+      return promisify(fs.access)(action.url, fs.constants.F_OK)
```

Comment:
You're already in an async context, why not just use `await`, and `try-catch` for handling the error?